src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -301,11 +301,19 @@ private function parseXmlAttributes(\DOMNode $node)
         $data = array();
 
         foreach ($node->attributes as $attr) {
-            if (ctype_digit($attr->nodeValue)) {
-                $data['@'.$attr->nodeName] = (int) $attr->nodeValue;
-            } else {
+            if (!is_numeric($attr->nodeValue)) {
```

Comment:
The current behavior is already unreliable and can break things. It's way I suggest to make this feature opt-in and disabled by default.